hr_expense_petty_cash/models/hr_expense.py

```diff
@@ -45,3 +45,11 @@ def _get_account_move_line_values(self):
             line['partner_id'] = expense.petty_cash_id.partner_id.id
             res[expense.id][1] = line
         return res
+
+    @api.multi
+    def _prepare_move_values(self):
+        self.ensure_one()
+        move_values = super()._prepare_move_values()
+        if self.payment_mode == 'petty_cash' and self.petty_cash_id.journal_id:
```

Comment:
I think we should not change the journal like this. It should be using the sheet.jounal_id to be correct.
But instead, when sheet is created, the petty cash's journal_id should be assigned.